src/pypdfium2/_helpers/document.py

```diff
@@ -80,15 +83,15 @@ def __init__(self, input, password=None, autoclose=False, to_close=None):
         if to_close:
             self._data_closer.extend(to_close)
         
-        self._input, to_close = _preprocess_input(self._orig_input)
+        input, to_close = _preprocess_input(self._orig_input)
```

Comment:
See https://github.com/pypdfium2-team/pypdfium2/pull/237#issuecomment-1642899392 for an idea how to keep input accessible on object level.